src/main/java/edu/cornell/kfs/sys/businessobject/lookup/CreateDoneBatchFileLookupableHelperServiceImpl.java

```diff
@@ -1,49 +1,56 @@
-import org.kuali.kfs.sys.batch.BatchFile;
-import org.kuali.kfs.sys.batch.BatchFileUtils;
-import org.kuali.kfs.kns.lookup.HtmlData;
-import org.kuali.kfs.kns.lookup.HtmlData.AnchorHtmlData;
-import org.kuali.rice.krad.bo.BusinessObject;
-import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.krad.bo.BusinessObjectBase;
 import org.kuali.kfs.krad.util.KRADConstants;
 import org.kuali.kfs.krad.util.UrlFactory;
+import org.kuali.kfs.sys.batch.BatchFile;
+import org.kuali.kfs.sys.batch.BatchFileUtils;
+import org.kuali.kfs.sys.businessobject.lookup.BatchFileLookupSearchServiceImpl;
+import org.kuali.rice.kim.api.identity.Person;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Properties;
+import edu.cornell.kfs.sys.batch.service.CreateDoneBatchFileAuthorizationService;
+
+public class CreateDoneBatchFileLookupableHelperServiceImpl extends BatchFileLookupSearchServiceImpl {
 
-public class CreateDoneBatchFileLookupableHelperServiceImpl extends CuBatchFileLookupableHelperServiceImpl {
-	
-	protected CreateDoneBatchFileAuthorizationService createDoneAuthorizationService; 
+	protected CreateDoneBatchFileAuthorizationService createDoneAuthorizationService;
 
 	@Override
-    public List<HtmlData> getCustomActionUrls(BusinessObject businessObject, List pkNames) {
-        List<HtmlData> links = new ArrayList<HtmlData>();
-        
-        BatchFile batchFile = (BatchFile) businessObject;
-        if (canCreateDoneFile(batchFile)) {
-            links.add(getCreateDoneUrl(batchFile));
-        }
-        return links;
-    }
-
-    protected boolean canCreateDoneFile(BatchFile batchFile) {
-    	boolean isDoneFile = StringUtils.endsWith(batchFile.getFileName(), ".done");
-    	return (!isDoneFile && createDoneAuthorizationService.canCreateDoneFile(batchFile, GlobalVariables.getUserSession().getPerson()));
-    }
-    
-    protected HtmlData getCreateDoneUrl(BatchFile batchFile) {
-        Properties parameters = new Properties();
-        parameters.put("filePath", BatchFileUtils.pathRelativeToRootDirectory(batchFile.retrieveFile().getAbsolutePath()));
-        parameters.put(KRADConstants.DISPATCH_REQUEST_PARAMETER, "createDone");
-        String href = UrlFactory.parameterizeUrl("../createDoneBatchFileAdmin.do", parameters);
-        return new AnchorHtmlData(href, "createDone", "Create Done");
-    }
-    
-    public CreateDoneBatchFileAuthorizationService getCreateDoneAuthorizationService() {
+	public List<Map<String, Object>> getActionLinks(BusinessObjectBase businessObject, Person user) {
+		BatchFile batchFile = (BatchFile) businessObject;
+		List<Map<String, Object>> actionLinks = new LinkedList<>();
+
+		if (canCreateDoneFile(batchFile, user)) {
+			Map<String, Object> createDoneLink = new LinkedHashMap<>();
+			createDoneLink.put("label", "Create Done");
+			createDoneLink.put("url", KRAD_URL_PREFIX + getCreateDoneUrl(batchFile));
+			createDoneLink.put("method", "GET");
```

Comment:
This class uses a lot of string literals for property names or standard property values. Can some or most of these be replaced with constants instead? (If needed, I'm okay with leaving them as-is if we intend to perform further cleanup on this file in a future sprint.)

Reply:
good point. I have modeled this on base code that that is not using any constants. I will add some constants to our local constants file.